ansible_collections/arista/avd/plugins/README.md

```diff
@@ -204,6 +210,142 @@ To use this filter:
+    {{ ip_pool | arista.avd.ipaddr }}
+    Output: ['192.168.0.1', 'fe08::1', '192.168.1.0/24']
+    ```
+
+    The below method is same as above.
+
+    ```example
+    ip_pool = ['192.168.0.1', '192.168.0', 'fe08::1', '192.168.1.0/24', '', True, 'test12.com']
+    {{ ip_pool | arista.avd.ipaddr('validate') }}
+    Output: ['192.168.0.1', 'fe08::1', '192.168.1.0/24']
+    ```
+
+1) `address`: Returns the valid IP address(es) only.
+
+    ```example
+    ip_pool = ['192.168.0.1', '192.168.0', 'fe08::1', '192.168.1.0/24', '', True]
+    {{ ip_pool | arista.avd.ipaddr('address') }}
+    Output: ['192.168.0.1', 'fe08::1']
+    ```
+
+1) `net`: Returns only the Network IPs.
+
+    ```example
+    ip_pool = ['192.168.0.1', '192.168.0', 'fe08::1', '192.168.1.0/24', '', True]
+    {{ ip_pool | arista.avd.ipaddr('net') }}
+    Output: ['192.168.1.0/24']
+    ```
+
+1) `subnet`: Calculates and returns the Network IP of the provided input.
+
+    ```example
+    ip_pool = ['192.168.0.1', 'fe08::1', '192.168.1.0/24', '192.168.1.20/28']
+    {{ ip_pool | arista.avd.ipaddr('subnet') }}
+    Output: ['192.168.0.1/32', 'fe08::1/128', '192.168.1.0/24', '192.168.1.16/28']
+    ```
+
+1) `netmask`: Calculates the subnet mask of the provided IP address(es).
+
+    ```example
+    ip_pool = ['192.168.0.1', 'fe08::1', '192.168.1.0/24', '192.168.0.70/26']
+    {{ ip_pool | arista.avd.ipaddr('netmask') }}
+    Output: ['255.255.255.255', 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff', '255.255.255.0', '255.255.255.192']
+    ```
+
+1) `prefix`: Calculates and returns the Network Prefix(es) for the provided IP address(es). Return value is an integer or list of integers.
+
+    ```example
+    ip_pool = ['192.168.0.1', 'fe08::1', '192.168.1.0/24', '192.168.0.4/26']
+    {{ ip_pool | arista.avd.ipaddr('prefix') }}
+    Output: [32, 128, 24, 26]
+    ```
+
+1) `size`: Calculates and returns the number of IP address(es) possible in the provided pool(s). Return value is an integer or list of integers.
+
+    ```example
+    ip_pool = ['192.168.0.1', 'fe08::1', '192.168.1.0/24', '192.168.0.4/26']
+    {{ ip_pool | arista.avd.ipaddr('size') }}
+    Output: [1, 1, 256, 64]
+    ```
+
+1) `network`: Calculates and returns the Network IP address(es).
+
+    ```example
+    ip_pool = ['192.168.0.1', 'fe08::1', '192.168.1.0/24', '192.168.0.70/26']
+    {{ ip_pool | arista.avd.ipaddr('network') }}
+    Output: ['192.168.0.1', 'fe08::1', '192.168.1.0', '192.168.1.64']
+    ```
+
+1) `defaultattr`: When the provided input method type doesn't with any of the above, then this method is invoked. Useful when the input method is not actually a method but an IP pool or a number. When the input argument is an integer then IP address corresponding to that number is returned. When an IP pool is provided as an argument then all of the input addresses which are subset of the IP pool are returned (last example).
```

Comment:
```suggestion
9) `defaultattr`: When the provided input method type doesn't match with any of the above, then this method is invoked. Useful when the input method is not actually a method but an IP pool or a number. When the input argument is an integer then IP address corresponding to that number is returned. When an IP pool is provided as an argument then all of the input addresses which are subset of the IP pool are returned (last example).
```

Reply:
Looks like this is not complete - from examples the argument can be an integer or an integer given as a string ?

Reply:
This method looks to be doing too many things at once with a very different outcome depending on the type or parseable type of the input - do we have use cases for this?